scripts/vm.sh

```diff
@@ -113,6 +105,58 @@ EOF
 	echo "Use: ssh $DEFAULT_USERNAME@$ip (password: $DEFAULT_PASSWORD)"
 }
 
+function is_gentoo() {
+	if uname -r | grep -iq "gentoo"; then
+		true
+	else
+		false
+	fi
+}
+
+function install_libs() {
+	if is_gentoo; then
+		echo "Gentoo system detected. Running emerge --sync..."
+		emerge --sync
+		emerge --deep app-emulation/libvirt app-emulation/qemu virtiofsd app-emulation/virt-manager whois
+		systemctl start libvirtd
+	else
+		echo "Non-Gentoo system detected. Running apt update..."
+		apt update
+		apt install -y libvirt-daemon-system virtiofsd
+	fi
+}
+
+function config_permissions() {
+	log "Configuring libvirt user permission and network"
+	if ! groups $USER | grep -q "\blibvirt\b"; then
+		usermod -a -G libvirt $USER
+	fi
+}
+
+function qemu_connect() {
+	if ! virsh --connect "qemu:///system" net-info default | grep -q "Active:.*yes"; then
+		virsh --connect "qemu:///system" net-start default
+	fi
+	if ! virsh --connect "qemu:///system" net-info default | grep -q "Autostart:.*yes"; then
+		virsh --connect "qemu:///system" net-autostart default
+	fi
+}
+
+function create_shared_folder() {
+	log "Creating shared folder"
+	mkdir -p "$SHARED_FOLDER"
+	chown -R ${SUDO_USER:-$USER}:libvirt "$SHARED_FOLDER"
+}
+
+function download_image() {
+	log "Downloading Debian cloud-ready image"
+
+	if is_gentoo; then
+		mkdir -p /var/lib/libvirt/images
+	fi
```

Comment:
This line is not required by Gentoo only and also could be improved to reuse the variable where the image path is stored. I prefer changing it to:
```suggestion
	mkdir -p $(dirname "$DEBIAN_QCOW2")
```